counsel.el

```diff
@@ -731,6 +731,25 @@ With a prefix arg, restrict list to variables defined using
       (when doc
         (lv-delete-window)))))
 
+;;** `counsel-tab-bar'
+(defun counsel--tab-name (tab)
+  "Return a name for a tab."
+  (cdr (car (cdr tab))))
+
+;;;###autoload
+(defun counsel-tab-bar ()
+  "Switch between tabs."
+  (interactive)
+  (ivy-read
+   "Select tab: "
+   (mapcar #'counsel--tab-name (funcall tab-bar-tabs-function))
```

Comment:
`tab-bar-tabs-function` and `tab-bar-switch-to-tab` are new in Emacs 27, so in older Emacsen they give byte-compiler warnings like the following:
```
In counsel-tab-bar:
counsel.el:745:41:Warning: reference to free variable ‘tab-bar-tabs-function’

In end of data:
counsel.el:6965:1:Warning: the following functions are not known to be
    defined: tab-bar-switch-to-tab
```
I think the best way to silence these is by adding the following after `(interactive)`:
```el
  (when (< emacs-major-version 27)
    (user-error "This command requires Emacs 27 or later"))
  (defvar tab-bar-tabs-function)
  (declare-function tab-bar-switch-to-tab "tab-bar" (name))
```